### PR TITLE
P1113R0 CWG "ready" Issues for the June, 2018 (Rapperswil) meeting

### LINKED ISSUE #2113: [2018-06 CWG Motion 1] P1113R0 CWG "ready" Issues for the June, 2018 (Rapperswil) meeting

### PR DESCRIPTION
**Pull request by tkoeppe**
FIxes #2113.

Note the comment in that issue.

### CODE REVIEW

File: source/templates.tex [hunk 1]
```diff
@@ -962,6 +962,45 @@
...
+function template specialization\iref{temp.deduct}.
+\begin{example}
+\begin{codeblock}
+template<class T, T::type n = 0> class X;
+struct S {
+  using type = int;
+};
+using T1 = X<S, int, int>;      // error: too many arguments
+using T2 = X<>;                 // error: no default argument for first template parameter
+using T3 = X<1>;                // error: value 1 does not match type-parameter
```
Comment (zygoloid) [inline source/templates.tex]: Missing code font for `1` here.

File: source/templates.tex [hunk 2]
```diff
@@ -7026,30 +7072,11 @@
...
-and no parameter is a template parameter pack, or if there is not
-an argument for each non-pack parameter,
-type deduction fails.
-
-\item If any non-type argument does not match the type of the
-corresponding non-type
-template parameter, and is not convertible to the type of the
-corresponding non-type parameter as specified in~\ref{temp.arg.nontype},
+When an explicit template argument list is specified, if the
+given \grammarterm{template-id} is invalid\iref{temp.names},
```
Comment (zygoloid) [inline source/templates.tex]: invalid -> not valid, to fix the use of an undefined term.

File: source/templates.tex [hunk 3]
```diff
@@ -7026,30 +7072,11 @@
...
 type deduction fails.
-
-\item The specified template argument values are substituted for the
+The specified template argument values are substituted for the
```
Comment (zygoloid) [inline source/templates.tex]: Add an "Otherwise" to the start of this sentence for clarity and flow.

File: source/templates.tex (hunk 1 above)
Comment (tkoeppe) [reply source/templates.tex]: Pushed as fixup.

File: source/templates.tex (hunk 3 above)
Comment (tkoeppe) [reply source/templates.tex]: ... and pushed as editorial commit.

**Pull request merged**